File: base/src/main/java/com/utils/Logger.java
package com.utils;

import android.util.Log;

/**
 * ka
 * 08/11/2017
 */

public class Logger {

    private static boolean isDebug = false;

    private final String TAG;

    private Logger(Class<?> clazz) {
        TAG = clazz.getSimpleName();
    }

    public static void init(boolean isDebug) {
        Logger.isDebug = isDebug;
    }

    public static Logger getLogger(Class<?> clazz) {
        return new Logger(clazz);
    }

    public void debug(Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.d(TAG, msg.toString());
            }
        }
    }

    public void debug(String tag, Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.d(tag, msg.toString());
            }
        }
    }

    public void debug(Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.d(TAG, msg.toString(), tr);
            } else {
                Log.d(TAG, "Exception", tr);
            }
        }
    }

    public void debug(String tag, Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.d(tag, msg.toString(), tr);
            } else {
                Log.d(tag, "Exception", tr);
            }
        }
    }

    public void info(Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.i(TAG, msg.toString());
            }
        }
    }

    public void info(String tag, Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.i(tag, msg.toString());
            }
        }
    }

    public void info(Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.i(TAG, msg.toString(), tr);
            } else {
                Log.i(TAG, "Exception", tr);
            }
        }
    }

    public void info(String tag, Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.i(tag, msg.toString(), tr);
            } else {
                Log.i(tag, "Exception", tr);
            }
        }
    }

    public void warn(Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.w(TAG, msg.toString());
            }
        }
    }

    public void warn(String tag, Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.w(tag, msg.toString());
            }
        }
    }

    public void warn(String tag, Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.w(tag, msg.toString(), tr);
            } else {
                Log.w(tag, "Exception", tr);
            }
        }
    }

    public void error(Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.e(TAG, msg.toString());
            }
        }
    }

    public void error(String tag, Object msg) {
        if (isDebug) {
            if (msg != null) {
                Log.e(tag, msg.toString());
            }
        }
    }

    public void error(Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.e(TAG, msg.toString(), tr);
            } else {
                Log.e(TAG, "Exception", tr);
            }
        }
    }

    public void error(String tag, Object msg, Throwable tr) {
        if (isDebug) {
            if (msg != null) {
                Log.e(tag, msg.toString(), tr);
            } else {
                Log.e(tag, "Exception", tr);
            }
        }
    }
}

File: base/src/main/java/com/utils/Logger.kt
package com.utils

import android.util.Log

/**
 * ka
 * 08/11/2017
 */

class Logger private constructor(clazz: Class<*>) {

    private val tag: String = clazz.simpleName

    fun debug(msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.d(tag, msg.toString())
            }
        }
    }

    fun debug(tag: String, msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.d(tag, msg.toString())
            }
        }
    }

    fun debug(msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.d(tag, msg.toString(), tr)
            } else {
                Log.d(tag, "Exception", tr)
            }
        }
    }

    fun debug(tag: String, msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.d(tag, msg.toString(), tr)
            } else {
                Log.d(tag, "Exception", tr)
            }
        }
    }

    fun info(msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.i(tag, msg.toString())
            }
        }
    }

    fun info(tag: String, msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.i(tag, msg.toString())
            }
        }
    }

    fun info(msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.i(tag, msg.toString(), tr)
            } else {
                Log.i(tag, "Exception", tr)
            }
        }
    }

    fun info(tag: String, msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.i(tag, msg.toString(), tr)
            } else {
                Log.i(tag, "Exception", tr)
            }
        }
    }

    fun warn(msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.w(tag, msg.toString())
            }
        }
    }

    fun warn(tag: String, msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.w(tag, msg.toString())
            }
        }
    }

    fun warn(tag: String, msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.w(tag, msg.toString(), tr)
            } else {
                Log.w(tag, "Exception", tr)
            }
        }
    }

    fun error(msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.e(tag, msg.toString())
            }
        }
    }

    fun error(tag: String, msg: Any?) {
        if (isDebug) {
            if (msg != null) {
                Log.e(tag, msg.toString())
            }
        }
    }

    fun error(msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.e(tag, msg.toString(), tr)
            } else {
                Log.e(tag, "Exception", tr)
            }
        }
    }

    fun error(tag: String, msg: Any?, tr: Throwable) {
        if (isDebug) {
            if (msg != null) {
                Log.e(tag, msg.toString(), tr)
            } else {
                Log.e(tag, "Exception", tr)
            }
        }
    }

    companion object {

        private var isDebug = false

        fun init(isDebug: Boolean) {
            Logger.isDebug = isDebug
        }

        fun getLogger(clazz: Class<*>): Logger {
            return Logger(clazz)
        }
    }
}
